### PR TITLE
Make the processing of sensor events safe

### PR DESCRIPTION
**Pull request by connorgorman**
## Description

Events from Sensor should be safe, would have prevented a previous panic in a handler

## Checklist
- [ ] Investigated and inspected CI test results
- [ ] Unit test and regression tests added
- [ ] Evaluated and added CHANGELOG entry if required
- [ ] Determined and documented upgrade steps
- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))

If any of these don't apply, please comment below.

## Testing Performed

Will purposefully cause panics in some handlers to check

### CODE REVIEW

File: central/sensor/service/connection/connection_impl.go [hunk 1]
```diff
@@ -155,8 +156,13 @@ func (c *sensorConnection) runRecv(ctx context.Context, grpcServer central.Senso
...
 	for msg := queue.pullBlocking(&c.stopSig); msg != nil; msg = queue.pullBlocking(&c.stopSig) {
-		if err := c.handleMessage(ctx, msg); err != nil {
-			log.Errorf("Error handling sensor message: %v", err)
+		err := safe.Run(func() {
+			if err := c.handleMessage(ctx, msg); err != nil {
+				log.Errorf("Error handling sensor message: %v", err)
+			}
+		})
+		if err != nil {
+			log.Errorf("UNEXPECTED panic in handle message: %v", err)
```
Comment (theencee) [inline central/sensor/service/connection/connection_impl.go:165]: Run already logs this (well technically `utils.ShouldErr`) and will panic on dev

File: central/sensor/service/connection/connection_impl.go [hunk 2]
```diff
@@ -155,8 +156,13 @@ func (c *sensorConnection) runRecv(ctx context.Context, grpcServer central.Senso
 
 func (c *sensorConnection) handleMessages(ctx context.Context, queue *dedupingQueue) {
 	for msg := queue.pullBlocking(&c.stopSig); msg != nil; msg = queue.pullBlocking(&c.stopSig) {
-		if err := c.handleMessage(ctx, msg); err != nil {
-			log.Errorf("Error handling sensor message: %v", err)
+		err := safe.Run(func() {
```
Comment (theencee) [inline central/sensor/service/connection/connection_impl.go:159]: why Run vs RunE?

File: central/sensor/service/connection/connection_impl.go (hunk 1 above)
Comment (connorgorman) [reply central/sensor/service/connection/connection_impl.go]: true, I removed the UNEXPECTED portion

File: central/sensor/service/connection/connection_impl.go (hunk 2 above)
Comment (connorgorman) [reply central/sensor/service/connection/connection_impl.go:159]: RunE does not allow me to distinguish between the types of errors and there are many times we have errors handling sensor messages

**Review by theencee — APPROVED** [review]

**Pull request merged**